src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -30,23 +30,27 @@ public class MeterProviderSdk
     {
         internal readonly ConcurrentDictionary<AggregatorStore, bool> AggregatorStores = new ConcurrentDictionary<AggregatorStore, bool>();
 
+        private readonly object collectLock = new object();
         private readonly CancellationTokenSource cts = new CancellationTokenSource();
         private readonly List<Task> collectorTasks = new List<Task>();
         private readonly MeterListener listener;
+        private readonly List<MeasurementProcessor> measurementProcessors = new List<MeasurementProcessor>();
+        private readonly List<MetricProcessor> metricProcessors = new List<MetricProcessor>();
 
         internal MeterProviderSdk(
             IEnumerable<string> meterSources,
             MeasurementProcessor[] measurementProcessors,
-            KeyValuePair<MetricProcessor, int>[] metricExportProcessors)
+            MetricProcessor[] metricProcessors)
         {
-            // Setup our Processors
+            this.measurementProcessors.AddRange(measurementProcessors);
+            this.metricProcessors.AddRange(metricProcessors);
 
-            this.MeasurementProcessors.AddRange(measurementProcessors);
-
-            this.ExportProcessors.AddRange(metricExportProcessors);
+            foreach (var processor in this.metricProcessors)
```

Comment:
Remember [the linked list](https://github.com/open-telemetry/opentelemetry-dotnet/blob/a9fed4252b2149a184e2a5fb855089fd11a39007/src/OpenTelemetry/CompositeProcessor.cs#L88)?

It might save the foreach heap allocation.

Reply:
yes. (this is at build time only, but yes will optimize)